osquery/core/system.cpp

```diff
@@ -471,4 +477,29 @@ DropPrivileges::~DropPrivileges() {
   }
 }
 #endif
+
+Status carvePaths(const std::set<std::string>& paths) {
+  auto guid = generateNewUUID();
+  pt::ptree tree;
+  tree.put("carve_guid", guid);
+  tree.put("time", getUnixTime());
+  tree.put("status", "STARTING");
+  tree.put("sha256", "");
+  tree.put("size", -1);
+  if (paths.size() > 1) {
+    tree.put("path", boost::algorithm::join(paths, ","));
+  } else {
+    tree.put("path", *(paths.begin()));
+  }
+  std::ostringstream os;
+  pt::write_json(os, tree, false);
+  auto s = setDatabaseValue(kCarveDbDomain, kCarverDBPrefix + guid, os.str());
+  if (!s.ok()) {
+    return s;
```

Comment:
It doesn't look like you leverage this return value anywhere. Perhaps let's just make this function a `void` and `VLOG(1)` if setting the database value fails?